src/index.js

```diff
@@ -48,427 +48,250 @@ const removeEventListener = (event) => {
-    if (isInitialized) {
-      switch (args.length) {
-      case 0:
-        AppLovinMAX.showInterstitial(adUnitId, null, null);
-        break;
-      case 1:
-        AppLovinMAX.showInterstitial(adUnitId, args[0], null);
-        break;
-      case 2:
-        AppLovinMAX.showInterstitial(adUnitId, args[0], args[1]);
-        break;
-      default:
-        // do nothing - unexpected number of arguments
-        break;
-      }
-    } else {
-      logUninitializedAccessError('showInterstitial');
-    }
-  });
+const showInterstitial = (adUnitId, ...params) => {
+  return runIfInit(showInterstitial.name,
+                   AppLovinMAX.showInterstitial,
+                   adUnitId, params[0], params[1]);
 };
 
 const setInterstitialExtraParameter = (adUnitId, key, value) => {
-  AppLovinMAX.isInitialized().then(isInitialized => {
-    if (isInitialized) {
-      AppLovinMAX.setInterstitialExtraParameter(adUnitId, key, value);
-    } else {
-      logUninitializedAccessError('setInterstitialExtraParameter');
-    }
-  });
+  return runIfInit(setInterstitialExtraParameter.name,
+                   AppLovinMAX.setInterstitialExtraParameter,
+                   adUnitId, key, value);
 }
 
 /*----------*/
 /* REWARDED */
 /*----------*/
 
 const loadRewardedAd = (adUnitId) => {
-  AppLovinMAX.isInitialized().then(isInitialized => {
-    if (isInitialized) {
-      AppLovinMAX.loadRewardedAd(adUnitId);
-    } else {
-      logUninitializedAccessError('loadRewardedAd');
-    }
-  });
+  return runIfInit(loadRewardedAd.name,
+                   AppLovinMAX.loadRewardedAd,
+                   adUnitId);
 }
 
 const isRewardedAdReady = (adUnitId) => {
-  return AppLovinMAX.isInitialized().then(isInitialized => {
-    if (isInitialized) {
-      return AppLovinMAX.isRewardedAdReady(adUnitId);
-    } else {
-      throw new Error("Failed to execute isRewardedAdReady() - please ensure the AppLovin MAX React Native module has been initialized by calling 'AppLovinMAX.initialize(...);'!");
-    }
-  });
+  return runIfInit(isRewardedAdReady.name,
+                   AppLovinMAX.isRewardedAdReady,
+                   adUnitId);
 }
 
-const showRewardedAd = (adUnitId, ...args) => {
-  AppLovinMAX.isInitialized().then(isInitialized => {
-    if (isInitialized) {
-      switch (args.length) {
-      case 0:
-        AppLovinMAX.showRewardedAd(adUnitId, null, null);
-        break;
-      case 1:
-        AppLovinMAX.showRewardedAd(adUnitId, args[0], null);
-        break;
-      case 2:
-        AppLovinMAX.showRewardedAd(adUnitId, args[0], args[1]);
-        break;
-      default:
-        // do nothing - unexpected number of arguments
-        break;
-      }
-    } else {
-      logUninitializedAccessError('showRewardedAd');
-    }
-  });
+const showRewardedAd = (adUnitId, ...params) => {
+  return runIfInit(showRewardedAd.name,
+                   AppLovinMAX.showRewardedAd,
+                   adUnitId, params[0], params[1]);
```

Comment:
Can you double-check that indexing `params` with `1` when you only have 1 argument does not crash? Technically we'd be indexing out of bounds. (same comment applies to other parts of this file, where applicable).

Reply:
Tested the following combinations and verified.

```
showRewardedAd(adUnitID, 'a');
showRewardedAd(adUnitID, 'a', 'b');
showRewardedAd(adUnitID, 'a', 'b',  'c');
showRewardedAd(adUnitID, 'a', null);
showRewardedAd(adUnitID, null, 'a');
showRewardedAd(adUnitID, null);
showRewardedAd(adUnitID, null, null);
```

Reply:
What about `showRewardedAd(adUnitID` or `showRewardedAd(adUnitID, 'a')`?

Reply:
Both also work.